Update Session User and RUM Global Context APIs (#15013)
DataDog/documentation: content/en/real_user_monitoring/browser/modifying_data_and_context.md

{{% /tab %}}
{{< /tabs >}}

## User session

> 

Author: ```suggestion
```

{{% /tab %}}
{{< /tabs >}}

## User session

> ### Identify user session

Reviewer: ```suggestion
```

Increase your filtering capabilities by adding extra attributes on top of the recommended ones. For instance, add information about the user plan, or which user group they belong to.

> When making changes to the user session object, all RUM events collected afterwards the change contain the updated information.

Reviewer 2: ```suggestion
When making changes to the user session object, all RUM events collected after the change contain the updated information.
```

> ## User session APIs

Reviewer: <!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about having a sub section for each capability (like global context)?
something like:
```
## user sessions
### identify user session
### access user session
### override user session property
### clear user session
```

Reviewer: it could be nice to merge 
```
## Identify user sessions
## User session APIs
```
into
```
## User session
```

it regroup all things related to user session in a single section, like Global Context

Increase your filtering capabilities by adding extra attributes on top of the recommended ones. For instance, add information about the user plan, or which user group they belong to.

When making changes to the user session object, all RUM events collected after the change contain the updated information.

> ### User session APIs

Reviewer: ```suggestion
### Identify user session
```

{{< tabs >}}
{{% tab "NPM" %}}
```
datadogRum.getUser()
```

{{% /tab %}}
{{% tab "CDN async" %}}
```
DD_RUM.onReady(function() {
    DD_RUM.getUser()
})
```
{{% /tab %}}
{{% tab "CDN sync" %}}

```
window.DD_RUM && window.DD_RUM.getUser()
```

{{% /tab %}}
{{< /tabs >}}

`datadogRum.setUserProperty('<USER_KEY>', <USER_VALUE>)`

Add/Override a property on the RUM Session User Object. 

{{< tabs >}}
{{% tab "NPM" %}}
```
datadogRum.setUserProperty('name', 'John Doe')
```

{{% /tab %}}
{{% tab "CDN async" %}}
```
DD_RUM.onReady(function() {
    DD_RUM.setUserProperty('name', 'John Doe')
})
```
{{% /tab %}}
{{% tab "CDN sync" %}}

```
window.DD_RUM && window.DD_RUM.setUserProperty('name', 'John Doe')
```

{{% /tab %}}
{{< /tabs >}}

`datadogRum.removeUserProperty('<USER_KEY>')`

{{< tabs >}}
{{% tab "NPM" %}}
```
datadogRum.removeUserProperty('name')
```

{{% /tab %}}
{{% tab "CDN async" %}}
```
DD_RUM.onReady(function() {
    DD_RUM.removeUserProperty('name')
})
```
{{% /tab %}}
{{% tab "CDN sync" %}}

```
window.DD_RUM && window.DD_RUM.removeUserProperty('name')
```

{{% /tab %}}
{{< /tabs >}}

`datadogRum.clearUser()`

Clear all properties from the RUM Session User object.

> > The RUM Browser SDK v4.17.0 introduced `clearUser` and deprecating `removeUser`

Reviewer 2: ```suggestion
<div class="alert alert-info">The RUM Browser SDK v4.17.0 introduced `clearUser` and deprecated `removeUser`</div>
```

{{% /tab %}}
{{< /tabs >}}

### Access user session

`datadogRum.getUser()`

> Get the current RUM Session User object.
> 

Author: ```suggestion
```

## Global context

### Add global context property

> Once RUM is initialized, add extra context to all RUM events collected from your application with the `setGlobalContextProperty(key: string, value: any)` API:

Reviewer 2: ```suggestion
After RUM is initialized, add extra context to all RUM events collected from your application with the `setGlobalContextProperty(key: string, value: any)` API:
```

Reviewer: ```suggestion
Add extra context to all RUM events collected from your application with the `setGlobalContextProperty(key: string, value: any)` API:
```
No need to wait for initialization

> > The RUM Browser SDK v4.17.0 introduced `setGlobalContextProperty` and deprecating `addRumGlobalContext`

Reviewer 2: ```suggestion
<div class="alert alert-info">The RUM Browser SDK v4.17.0 introduced `setGlobalContextProperty` and deprecated `addRumGlobalContext`</div>
```

Follow the [Datadog naming convention][16] for a better correlation of your data across the product.

### Remove global context property

You can remove a previously defined global context property.

> > The RUM Browser SDK v4.17.0 introduced `removeGlobalContextProperty` and deprecating `removeRumGlobalContext`

Reviewer 2: ```suggestion
<div class="alert alert-info">The RUM Browser SDK v4.17.0 introduced `removeGlobalContextProperty` and deprecated `removeRumGlobalContext`</div>
```

{{< tabs >}}
{{% tab "NPM" %}}

```
import { datadogRum } from '@datadog/browser-rum';

datadogRum.removeGlobalContextProperty('<CONTEXT_KEY>');

// Code example
datadogRum.removeGlobalContextProperty('activity');
```

{{% /tab %}}
{{% tab "CDN async" %}}
```
DD_RUM.onReady(function() {
    DD_RUM.removeGlobalContextProperty('<CONTEXT_KEY>');
})

// Code example
DD_RUM.onReady(function() {
    DD_RUM.removeGlobalContextProperty('activity');
})
```
{{% /tab %}}
{{% tab "CDN sync" %}}

```
window.DD_RUM && window.DD_RUM.removeGlobalContextProperty('<CONTEXT_KEY>');

// Code example
window.DD_RUM && window.DD_RUM.removeGlobalContextProperty('activity');
```

{{% /tab %}}
{{< /tabs >}}

Follow the [Datadog naming convention][16] for a better correlation of your data across the product.

### Replace global context

> Once RUM is initialized, replace the default context for all your RUM events with the `setGlobalContext(context: Context)` API:

Reviewer: ```suggestion
Replace the default context for all your RUM events with the `setGlobalContext(context: Context)` API:
```

> > The RUM Browser SDK v4.17.0 introduced `setGlobalContext` and deprecating `setRumGlobalContext`

Reviewer 2: ```suggestion
<div class="alert alert-info">The RUM Browser SDK v4.17.0 introduced `setGlobalContext` and deprecated `setRumGlobalContext`</div>
```

Follow the [Datadog naming convention][16] for a better correlation of your data across the product.

### Clear global context

> You can reset the global context by using `clearGlobalContext`.

Reviewer: ```suggestion
You can clear the global context by using `clearGlobalContext`.
```

{{< tabs >}}
{{% tab "NPM" %}}

```
import { datadogRum } from '@datadog/browser-rum';

datadogRum.clearGlobalContext();

// Code example
datadogRum.clearGlobalContext();
```

{{% /tab %}}
{{% tab "CDN async" %}}
```
DD_RUM.onReady(function() {
    DD_RUM.clearGlobalContext();
})

// Code example
DD_RUM.onReady(function() {
    DD_RUM.clearGlobalContext()
})
```
{{% /tab %}}
{{% tab "CDN sync" %}}

```
window.DD_RUM &&
    DD_RUM.clearGlobalContext();

// Code example
window.DD_RUM &&

>     DD_RUM.clearGlobalContext();

Reviewer 3: <!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏remove "code examples" as they are the same as the above

```

{{% /tab %}}
{{< /tabs >}}

> Follow the [Datadog naming convention][16] for a better correlation of your data across the product.

Reviewer: ```suggestion
```
not really relevant here

### Read global context

Once RUM is initialized, read the global context with the `getGlobalContext()` API:

> > The RUM Browser SDK v4.17.0 introduced `getGlobalContext` and deprecating `getRumGlobalContext`

Reviewer 2: ```suggestion
<div class="alert alert-info">The RUM Browser SDK v4.17.0 introduced `getGlobalContext` and deprecated `getRumGlobalContext`</div>
```
